data_model/src/role.rs

```diff
@@ -81,11 +92,11 @@ impl Role {
     /// Constructor.
     #[inline]
     pub fn new(
-        id: impl Into<Id>,
+        id: <Self as Identifiable>::Id,
```

Comment:
if you want to make this change, I think there are other places in `data_model` where the same form of constructor was used

Reply:
I did it so it can be consistent with other sturctures like `Accoount`, `Domain` and etc.
Also there seems to be nothing `role::Id` can be constructed with using `From::from()` except itself.

I can't find anything like that in `data_model`. I think you eliminated such thing in your refactoring PR